src/main/java/pixel/util/Parser.java

```diff
@@ -0,0 +1,142 @@
+                int indexToChange = Character.getNumericValue(temp.charAt(0));
+                // System.out.println(indexToChange);
+                if ((indexToChange > 0) && (indexToChange < 100)) {
+                    Storage.INPUT_TASKS.get(indexToChange - 1).markAsDone();
+                }
+
+                Storage.resetFile(this.filePath);
+                for (Task task : Storage.INPUT_TASKS) {
+                    Storage.appendToFile(task, this.filePath);
+                }
+
+                System.out.println(" Nice! I've marked this task as done:");
+                System.out.println(Storage.INPUT_TASKS.get(indexToChange - 1));
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().startsWith("unmark ")) {
+                // truncate the front part
+                String temp = userInput.substring(7);
+                // System.out.println(temp);
+                int indexToChange = Character.getNumericValue(temp.charAt(0));
+                // System.out.println(indexToChange);
+                if ((indexToChange > 0) && (indexToChange < 100)) {
+                    Storage.INPUT_TASKS.get(indexToChange - 1).markAsNotDone();
+                }
+
+                Storage.resetFile(this.filePath);
+                for (Task task : Storage.INPUT_TASKS) {
+                    Storage.appendToFile(task, this.filePath);
+                }
+
+                System.out.println("OK, I've marked this task as not done yet:");
+                System.out.println(Storage.INPUT_TASKS.get(indexToChange - 1));
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().equals("list")) {
+                // System.out.println(inputMemory.length);
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < Pixel.count; i++) {
+                    Task currentTask = Storage.INPUT_TASKS.get(i);
+                    System.out.println((i + 1) + ". " + currentTask);
+                }
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().startsWith("delete ")) {
+                Storage.deleteEntry(userInput, filePath);
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else if (userInput.strip().startsWith("find ")) {
+                ArrayList<Task> results = Storage.findEntry(userInput);
+                System.out.println("Here are the matching tasks in your list:");
+                for (int i = 0; i < results.size(); i++) {
+                    Task currentTask = results.get(i);
+                    System.out.println((i + 1) + ". " + currentTask);
+                }
+                System.out.println(UserInterface.AFTER_VALID_INPUT);
+
+            } else {
+                throw new IncorrectFormatException("Input should be a task or command!"); // programme breaks
+            }
```

Comment:
I like that you separated by one blank line for each logical units within a block, make it easier to read.